### PR TITLE
feat: [SPE-1491] Prepare ruby

### PR DESCRIPTION
**Pull request by idbentley**

### CODE REVIEW

**Review by simplesagar — APPROVED** [review]

File: .github/workflows/sdk-generation.yaml [hunk 1]
```diff
@@ -190,6 +195,8 @@ jobs:
       java_directory: ${{ steps.generate.outputs.java_directory }}
       php_regenerated: ${{ steps.generate.outputs.php_regenerated }}
       php_directory: ${{ steps.generate.outputs.php_directory }}
+      ruby_regenerated: ${{ steps.generate.outputs.php_regenerated }}
```
Comment (TristanSpeakEasy) [inline .github/workflows/sdk-generation.yaml:198]:
```suggestion
      ruby_regenerated: ${{ steps.generate.outputs.ruby_regenerated }}
```

File: .github/workflows/sdk-generation.yaml [hunk 2]
```diff
@@ -190,6 +195,8 @@ jobs:
...
+      ruby_directory: ${{ steps.generate.outputs.php_directory }}
```
Comment (TristanSpeakEasy) [inline .github/workflows/sdk-generation.yaml:199]:
```suggestion
      ruby_directory: ${{ steps.generate.outputs.ruby_directory }}
```

File: .github/workflows/sdk-generation.yaml
Comment (TristanSpeakEasy) [inline .github/workflows/sdk-generation.yaml:1]: you will want to add a `compile-ruby` step to here and add the check into the `finalize` step

File: action.yml
Comment (TristanSpeakEasy) [inline action.yml:1]: `publish_ruby` definition needs to be added to this file

Comment (TristanSpeakEasy) [reply action.yml:1]: and the inputs for ruby into the docker image at the bottom

File: pkg/releases/releases.go [hunk 3]
```diff
@@ -188,7 +192,7 @@ func ParseReleases(data string) (*ReleasesInfo, error) {
 		}
 
 		info.Languages["go"] = LanguageReleaseInfo{
-			Version:     goMatches[1],
+			Version:     goMatches[1],
```
Comment (TristanSpeakEasy) [inline pkg/releases/releases.go:195]:
```suggestion
			Version:     goMatches[1],
```

File: .github/workflows/sdk-generation.yaml
Comment (TristanSpeakEasy) [reply .github/workflows/sdk-generation.yaml:1]: there is also [sdk-publish.yaml](https://github.com/speakeasy-api/sdk-generation-action/blob/main/.github/workflows/sdk-publish.yaml) that needs to be modified

**Review by TristanSpeakEasy — CHANGES_REQUESTED** [review]
Changes required:

- [x] readme will need updating with new settings
- [x] ruby details need to be added to [action-inputs-config.json](https://github.com/speakeasy-api/sdk-generation-action/blob/main/.github/action-inputs-config.json)
- [x] ruby details need adding to [action-security-config.json](https://github.com/speakeasy-api/sdk-generation-action/blob/main/.github/action-security-config.json)
- [x] in `generate.go` if ruby supports installing packages from git directly (instead of a package manager), `getInstallationURL` needs a ruby case for returning a git installation URL
- [x] update `.github/workflows/sdk-generation.yaml` with publish-ruby job and ruby support
- [x] plus the other comments I left

File: .github/workflows/sdk-generation.yaml
Comment (TristanSpeakEasy) [reply .github/workflows/sdk-generation.yaml:1]: will also need a `publish_gems` or similar step in here and the sdk-publish.yaml see `publish-packagist` for how its done for PHP

File: .github/workflows/sdk-generation.yaml [hunk 4]
```diff
@@ -391,6 +399,32 @@ jobs:
...
+    needs: generate
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.generate.outputs.branch_name }}
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@ec02537da5712d66d4d50a0f33b7eb52773b5ed1
+        with:
+          ruby-version: "3.1"
+      - run: bundle install
```
Comment (TristanSpeakEasy) [inline .github/workflows/sdk-generation.yaml:415]: generally the steps here we will want to mirror the steps in the `compile.js` file from the templates ie `gem build && bundle install && rake rubocop:autocorrect` as we want this step to fail if the generator outputs invalid ruby

File: .github/workflows/sdk-publish.yaml [hunk 5]
```diff
@@ -268,3 +279,43 @@ jobs:
           footer: "Linked Repo <{repo_url}|{repo}> | <{run_url}|View Run>"
         env:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+  publish-gems:
+    if: ${{ needs.generate.outputs.ruby_regenerated == 'true' && inputs.publish_ruby == 'true' }}
```
Comment (TristanSpeakEasy) [inline .github/workflows/sdk-publish.yaml:283]:
```suggestion
    if: ${{ needs.release.outputs.ruby_regenerated == 'true' && inputs.publish_ruby == 'true' }}
```

File: .github/workflows/sdk-publish.yaml [hunk 6]
```diff
@@ -268,3 +279,43 @@ jobs:
...
+        working-directory: ${{ needs.generate.outputs.ruby_directory }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.finalize.outputs.commit_hash }}
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@ec02537da5712d66d4d50a0f33b7eb52773b5ed1
+        with:
+          ruby-version: "3.1"
+        working-directory: ${{ needs.generate.outputs.ruby_directory }}
```
Comment (TristanSpeakEasy) [inline .github/workflows/sdk-publish.yaml:298]:
```suggestion
        working-directory: ${{ needs.release.outputs.ruby_directory }}
```

File: .github/workflows/sdk-publish.yaml [hunk 7]
```diff
@@ -268,3 +279,43 @@ jobs:
...
+    if: ${{ needs.generate.outputs.ruby_regenerated == 'true' && inputs.publish_ruby == 'true' }}
+    name: Publish Ruby SDK
+    runs-on: ubuntu-latest
+    needs: release
+    defaults:
+      run:
+        working-directory: ${{ needs.generate.outputs.ruby_directory }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
```
Comment (TristanSpeakEasy) [inline .github/workflows/sdk-publish.yaml:292]: the `with` section is needed here

File: README.md [hunk 8]
```diff
@@ -176,14 +176,14 @@ Publishing a generated terraform provider is possible through the configuration
...
-  1. The GPG private key.
-  2. The GPG passphrase.
-  3. The GPG Public Key
-3. Add the ASCII-armored public key to the terraform registry.
-4. Ensure that the following secrets are available to your repository. These will be configured automatically if entered into the speakeasy UI.
-  1. `TERRAFORM_GPG_PRIVATE_KEY`: The GPG private key.
-  2. `TERRAFORM_GPG_PASSPHRASE`: The GPG passphrase.
-5. Once the initial release of your provider is complete (after executing this github action), you will need to manually add the provider to the terraform registry. Follow [the terraform registry instructions](https://registry.terraform.io/publish/provider) to begin this process, and agree to any Terraform Terms and Conditions. You will need to be an organizational admin to complete this step. This step needs only be performed once: subsequent updates will happen automatically.
+3. The GPG private key.
```
Comment (TristanSpeakEasy) [inline README.md:179]: might want to revert the changes to this list as I think some of these are meant to be indented as per the original

File: README.md [hunk 9]
```diff
@@ -293,6 +294,11 @@ This will also create a tag for the release, allowing the Go SDK to be retrieved
 **(Workflow Only)** Whether to publish the PHP SDK for Composer. Default `"false"`.
 **Note**: Needs to be set in the generate and publish workflows if using `pr` mode.
 
+### `publish_ruby`
+
+**(Workflow Only)** Whether to publish the Ruby SDK to Rubygems. Default `"flase"`
```
Comment (TristanSpeakEasy) [inline README.md:299]:
```suggestion
**(Workflow Only)** Whether to publish the Ruby SDK to Rubygems. Default `"false"`
```

File: .github/workflows/sdk-publish.yaml (hunk 7 above)
Comment (idbentley) [reply .github/workflows/sdk-publish.yaml:292]: I don't understand what you mean by this?

File: .github/workflows/sdk-publish.yaml [hunk 10]
```diff
@@ -268,3 +279,41 @@ jobs:
...
+    defaults:
+      run:
+        working-directory: ${{ needs.generate.outputs.ruby_directory }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@ec02537da5712d66d4d50a0f33b7eb52773b5ed1
+        with:
+          ruby-version: "3.1"
+        working-directory: ${{ needs.release.outputs.ruby_directory }}
```
Comment (TristanSpeakEasy) [inline .github/workflows/sdk-publish.yaml:296]: this line `working-directory: ${{ needs.release.outputs.ruby_directory }}` can be omitted as you are using the defaults above

**Review by TristanSpeakEasy — APPROVED** [review]
LGTM let me know when you are ready and I will get this merged, released and let you know when its ready for testing
